src/v5/attached.rs

```diff
@@ -0,0 +1,143 @@
+use bytes::{Buf, Bytes};
+use protobuf::Message;
+
+use crate::{aes::IvAndCiphertext, icl_header_v4::V4DocumentHeader, Error};
+
+use super::key_id_header::{self, KeyIdHeader};
+
+type Result<A> = std::result::Result<A, Error>;
+
+#[derive(Debug, PartialEq)]
+pub struct AttachedDocument {
+    pub key_id_header: KeyIdHeader,
+    pub edek: V4DocumentHeader,
+    pub edoc: IvAndCiphertext,
+}
+
+/// Construct an IronCore attached EDOC from the constituent parts.
+pub fn encode_attached_edoc(
+    AttachedDocument {
+        key_id_header,
+        edek,
+        edoc,
+    }: &AttachedDocument,
+) -> Result<Bytes> {
+    let key_id_header_bytes = key_id_header.write_to_bytes();
+    let encoded_edek = edek.write_to_bytes().expect("Writing to bytes is safe");
+    if encoded_edek.len() > u16::MAX as usize {
+        Err(Error::HeaderLengthOverflow(encoded_edek.len() as u64))
+    } else {
+        let len = encoded_edek.len() as u16;
+
+        let result = [
+            key_id_header_bytes.as_ref(),
+            &len.to_be_bytes(),
+            &encoded_edek,
+            &edoc.0, // Note that the edoc is written without a header since it's an attached document.
```

Comment:
By "without a header" do you mean "0IRON"? It's a bit confusing because this does have a V4DocumentHeader right before it.